pkg/api/controller.go

```diff
@@ -1611,7 +1611,8 @@ func handleAPIError(w http.ResponseWriter, err error) bool {
 	switch {
 	case errors.Is(err, catalog.ErrNotFound),
 		errors.Is(err, graveler.ErrNotFound),
-		errors.Is(err, actions.ErrNotFound):
+		errors.Is(err, actions.ErrNotFound),
+		errors.Is(err, db.ErrNotFound):
```

Comment:
```suggestion
		errors.Is(err, actions.ErrNotFound),
		errors.Is(err, auth.ErrNotFound),
		errors.Is(err, db.ErrNotFound):
```
(and the same below).  The error returned by `auth` just happens to be an error from package `db`, but "obviously" it will be that error but from package `auth`.  In fact this is probably wrong; probably a different identical type should be returned by package `auth`.  So `errors.go` should say
```go
var (
	ErrNotFound                = fmt.Sprintf("%w", db.ErrNotFound)
	ErrAlreadyExists           = fmt.Sprintf("%w", db.ErrAlreadyExists)
// ...
)
```
or something, and not
```go
var (
	ErrNotFound                = db.ErrNotFound
	ErrAlreadyExists           = db.ErrAlreadyExists
// ...
)
```
.

Alternatively, _everything_ should wrap a separate "base" error `ErrNotFound`, and then a single `errors.Is` here would suffice.

None of this errorefactoring is needed here (of course).  But I would like you to catch errors from the package(s) that we called to generate them, in this case `auth`.

Reply:
I'll open a separate issue with a suggestion to refactor the error types and possibly handling